File: BackEnd/src/CollectFlow.Domain/Entities/AdminUser.cs
using CollectFlow.Domain.Enums;

namespace CollectFlow.Domain.Entities;

public class AdminUser : BaseEntity
{
    public Guid TenantId { get; set; }
    public Tenant Tenant { get; set; } = default!;

    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PasswordHash { get; set; } = string.Empty;
    public AdminUserRole Role { get; set; } = AdminUserRole.Admin;
    public bool IsActive { get; set; } = true;
    public DateTime? LastLoginAtUtc { get; set; }
}

File: FrontEnd/src/types/adminUser.ts
export type AdminUserRole = 'Admin' | 'Sales' | 'Operations' | 'Viewer' | number;

export type AdminUser = {
  id: string;
  fullName: string;
  email: string;
  role: AdminUserRole;
  isActive: boolean;
  createdAtUtc: string;
  lastLoginAtUtc?: string | null;
};

export type CreateAdminUserRequest = {
  fullName: string;
  email: string;
  password: string;
  role: string;
};
